File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr = head
        while curr:
            nextNode = curr.next
            while nextNode and nextNode.val == curr.val:
                curr.next = nextNode.next
                nextNode = nextNode.next

            curr = curr.next

        return head

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let curr = head;
    while (curr) {
        let nextNode = curr.next;
        while (nextNode && nextNode.val === curr.val) {
            curr.next = nextNode.next;
            nextNode = nextNode.next;
        }
        curr = curr.next;
    }
    return head;
};